### ISSUE TITLE
I wish Eleventy could transform more file types

### ISSUE THREAD

**Issue by Navigatron** [11ty/eleventy#1015]
I have spent the last few hours working to implement `less` &rarr; `css` pre-processing in my build process. There is no plugin - but with transforms, I figured I would take a crack at it.

I had hoped that this would work:

```js
eleventyConfig.addTransform(
	"compile-my-less",
	function(content, outputPath) {
		if(outputPath.endsWith(".less")){
			// This is where I would perform the transformation
			return content;
		}

		// return this file unchanged
		return content;
	}
);
```

However, there are two problems.

According to [the pass-through documentation](https://www.11ty.dev/docs/copy/), eleventy searches for files based on the extensions in the `templateFormats` array.

> If a file format is not recognized by Eleventy as a valid template file extension, Eleventy will simply copy this file directly to your output.

What I have learned is that this means my `less` files, or any file other than those listed in the [template languages section](https://www.11ty.dev/docs/languages/), will not hit any transforms.

*The second, smaller issue - I can't change the output path in the transform, less files would have to use the css extension, and the transform would have to apply to all css - but that's not as big of a deal.*

I'm not looking for eleventy to implement `less` rendering - I'm looking for a way so that I can implement it, and so other people can implement processing on any other file extension.

Please consider calling transforms on all files. Or, worst case, have unknownTransforms for unknown files.

**Comment by octoxalis** [comment]
Don't use transform but filter it's all you need.

**Comment by pdehaan** [comment]
FWIW, we were trying to get **postcss** to compile via custom filters and shortcodes in #1017 and that might be a similar solution for compiling .less templates.

**Comment by zachleat** [comment]
I don’t think we want to change transforms for this. It needs to be implemented in the correct way (via #117), which is in testing mode now!

Going to close this as a duplicate—it’s coming!

If you want something now, I’d recommend using 11ty.js templates to work around this for now. You can transform whatever you want in those and output it a `.css` file using `permalink`.

**Comment by pspeter3** [comment]
Is #117 close to launching? It does seem like supporting simple transforms on pass through files could be nice? (Eg, just minifying some CSS or JS)
